main.tf

```diff
@@ -56,8 +56,8 @@ locals {
   worker_specs = {
     for pool in var.worker_pools :
     pool.pool_name => {
-      cpu_count = tonumber(split("x", split(".", pool.machine_type)[1])[0])
-      ram_count = tonumber(split("x", split(".", pool.machine_type)[1])[1])
+      cpu_count = can(regex("^.*?(\\d+)x(\\d+)", pool.machine_type)) ? tonumber(regex("^.*?(\\d+)x(\\d+)", pool.machine_type)[0]) : 0
```

Comment:
In the odd chance that the `can()` function fails the regex, you will set these values to "0", what would be the result of that for the worker pool? Is that ok to happen?

If not, then we might need some validation for the machine_type input variable to stop the user before it gets this far.

Reply:
I have now included a validation for machine_type and modified the validation required to get CPU and RAM values by removing use of `can`. Could you please check it once again @toddgiguere ?